frame/support/procedural/src/construct_runtime/expand/inherent.rs

```diff
@@ -90,9 +90,37 @@ pub fn expand_outer_inherent(
 				use #scrate::inherent::{ProvideInherent, IsFatalError};
 				use #scrate::traits::{IsSubType, ExtrinsicCall};
 				use #scrate::sp_runtime::traits::Block as _;
+				use #scrate::_private::sp_inherents::Error;
+				use #scrate::log;
 
 				let mut result = #scrate::inherent::CheckInherentsResult::new();
 
+				// This handle assume we abort on the first fatal error.
+				fn handle_put_error_result(res: Result<(), Error>) {
+					const LOG_TARGET: &str = "runtime::inherent";
+					match res {
+						Ok(()) => (),
+						Err(Error::InherentDataExists(id)) =>
+							log::debug!(
+								target: LOG_TARGET,
+								"Some error already reported for inherent {:?}, new non fatal \
+								error is ignored",
+								id
+							),
+						Err(Error::FatalErrorReported) =>
+							log::error!(
+								target: LOG_TARGET,
+								"Fatal error already reported, unexpected considering there is \
+								only one fatal error",
+							),
+						Err(Error::DecodingFailed(_, _)) | Err(Error::Application(_)) =>
+							log::error!(
```

Comment:
Could use `defensive` macro here instead, as we still know these should (almost) never happen.

Reply:
but `defensive` doesn't support renaming of `frame_support` contrary to `construct_runtime`.

(by renaming mean if somebody rename in its Cargo.toml like this:
```
s = { version = "*", package = "frame-support" }
```
)